packages/playwright-test/src/expect.ts

```diff
@@ -92,7 +92,40 @@ export const printReceivedStringContainExpectedResult = (
 
 // #endregion
 
-export const expect: Expect = expectLibrary as any;
+class ExpectMetaInfoProxy {
+  private _message: string;
+
+  constructor(message: string) {
+    this._message = message;
+  }
+
+  get(target: any, prop: any, receiver: any): any {
+    const value = Reflect.get(target, prop, receiver);
+    if (typeof value !== 'function')
+      return new Proxy(value, this);
+    return (...args: any[]) => {
+      try {
+        expectCallMetaInfo = {
+          message: this._message,
+        };
+        const result = value.call(target, ...args);
+        return result;
+      } finally {
+        expectCallMetaInfo = undefined;
+      }
+    };
+  }
+}
+
+export const expect: Expect = new Proxy(expectLibrary as any, {
+  apply: function(target: any, thisArg: any, argumentsList: [actual: unknown, message: string|undefined]) {
+    const message: string = argumentsList[1] || '';
+    if (typeof message !== 'string')
+      throw new Error('Second `expect` argument must be string!');
```

Comment:
`expect(actual, optionalErrorMessage): optional error message must be a string.`